active/001-tabular-msa.md

```diff
@@ -0,0 +1,335 @@
+no validation and a (slightly) crippled API.
+
+In order to better reflect the non-biological nature of `Sequence` we also
+recommend changing the `repr` to be different from `IUPACSequence`.
+
+Adding `base_frequencies()` as a method to `Sequence` will improve the value of
+returning `Sequence` object for the column.
+
+# Drawbacks
+
+- scikit-bio users will have to update their code to use a new API.
+
+- The new class name `TabularMSA` may not be as obvious as to its behavior as
+the current class name `Alignment`.
+
+- It is unclear if it is advisable to use a tabular (row-column) MSA as it
+forces an ordering where none exists. See
+[Lee et al. 2002](http://bioinformatics.oxfordjournals.org/content/18/3/452.short).
+
+# Alternatives
+
+We considered representing an MSA internally as a DAG in support of
+partial-order MSAs. However, representing positions in such an MSA defeats the
+entire purpose of using a DAG at all.
+
+We considered *only* supporting partial-order MSAs. However, since scikit-bio
+is a bioinformatics data munging library, we need to support row-column MSAs
+because most bioinformatics tools use or produce these structures.
+
+We considered the following alternative names for the new class:
+
+- `LinearMSA`: it is not immediately obvious that this is a row-column
+alignment
+- `TotalOrderMSA`: even less obvious and it sounds "better" than partial-order
+if users aren't clear on the definitions
+- `RowColMSA` / `RowColumnMSA`: painful to type and hard to remember (was it
+"col" or "column"?)
+- `StandardAlignment`: implies "best" or preferred alignment structure
+- `Alignment`: too general as it doesn't describe an MSA, also implies a
+default alignment structure
+- `MatrixMSA`: the name is ambiguous and could be confused with other alignment
+structures such as an alignment scoring matrix. It could also be confused with
+MSAs represented by graphs because graphs can be stored as matrices (e.g.,
+adjacency, incidence). Does not indicate a labeled structure. May imply too
+much about the internal implementation.
+
+# Unresolved questions
+
+What does the interface of `merge` look like? Can both axes be merged?
+If so, do we support the standard SQL join operations? Should it be called
+`join` instead of `merge`?
+
+How should non-standard characters be handled when computing entropy? Currently
+`Alignment's` method `position_entropies` has a keyword
+`nan_on_non_standard_chars`. This seems like a sensible option, however with
+our new boolean vector filtering model, users may be unaware that `nan` will
+always return `False` for any boolean operation. This may result in
+unanticipated behavior depending on how the boolean array was constructed. For
+example, `(ents <= 0.8) & (ents>= 0.3)` is semantically identical to
+`~((ents > 0.8) | (ents < 0.3))` but will handle `nan` in opposite ways. Is it
+acceptable to keep the current behavior or is there a better way to handle
+non-standard characters when computing entropy?
```

Comment:
Other methods that have been developed since the paper @gregcaporaso linked to:

- [FRcons](http://bioinformatics.oxfordjournals.org/content/24/5/613.full)
- [Jensen-Shannon divergence](http://bioinformatics.oxfordjournals.org/content/23/15/1875.full)
- [ConSurf](http://nar.oxfordjournals.org/content/33/suppl_2/W299.short)

These papers also contain surveys/comparisons to existing conservation algorithms (e.g., Shannon entropy).

I don't think there is any discussion about how to handle degenerate characters, but the papers touch on other important considerations when computing conservation:

- how to handle gaps
- normalizing to account for sequence redundancy (sequence weighting)

Since there are many ways to compute conservation (Shannon entropy being one of them) we should account for this in the API design. Perhaps a `TabularMSA.conservation` method that supports a number of metrics? This would match how we're handling pairwise distance calculations in `skbio.diversity.beta.pw_distances`.